test/e2e/set/nodeconfig/nodeconfig_disksetup.go

```diff
@@ -281,8 +282,18 @@ var _ = g.Describe("Node Setup", framework.Serial, func() {
 
 		// Disable disk setup before cleanup to not fight over resources
 		// TODO: can be removed once we support cleanup of filesystem and raid array
-		nc.Spec.LocalDiskSetup = nil
-		nc, err = f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Update(ctx, nc, metav1.UpdateOptions{})
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			var freshNC, updatedNC *scyllav1alpha1.NodeConfig
+			freshNC, err = f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Get(ctx, nc.Name, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			freshNC.Spec.LocalDiskSetup = nil
+			updatedNC, err = f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Update(ctx, freshNC, metav1.UpdateOptions{})
+			if err == nil {
+				nc = updatedNC
+			}
+			return err
```

Comment:
nit: imo it's clearer to return err explicitly, but we don't have a coding convention yet so feel free to ignore

Reply:
I think this is a bit special from the regular cases - you want to return the unmodified error whether it is nil an non-nil and no code can follow up after that.

Reply:
> I am also not exactly sure what's proposed here? I assumed this was return nil

Yup, I meant a block for err != nil and return nil later.

> you want to return the unmodified error whether it is nil or non-nil and no code can follow up after that

fair enough, thanks